litgpt/finetune/adapter.py

```diff
@@ -294,13 +291,20 @@ def fit(
                 save_hyperparameters(setup, checkpoint_file.parent)
                 save_prompt_style(data.prompt_style, checkpoint_file.parent)
 
+        if step_count == max_steps:
```

Comment:
You would want to also check if you already validated this step

Reply:
this was a bit tricky ... the least convoluted way for this was to introduce a new variable because otherwise you have to do a complicated check since the model is not validated after each step.